sysdeps/nacl/ioctl.c

```diff
@@ -11,10 +11,18 @@ int __ioctl (int fd, unsigned long request, ...) {
       return -1;
     }
 
+    /*In requests that do not have any arguments, we simply pass a pointer to the int
+      we define here as a placeholder since Lind excepts a pointer at the NaCl side.*/
     int arg = 0;
-    void* arg_ptr = &arg;
+    void* arg_ptr = &arg; //Replaced later for requests using an arg
     int result;
 
+    /*Right now, we are only passing down pointers to NaCl. Any non-pointer value
+      for arg_ptr will cause a sysaddr translation error in NaCl.
+      As a result, the arg_ptr should be set to &va_arg (ap, void *)
+      instead of va_arg (ap, void *) for requests that directly accept 
+      an arg value and not a pointer.*/
```

Comment:
Also mention that this should be the case for struct pointers too.

Reply:
How does this apply to struct pointers? Thought va_arg (ap, void *) would be valid for struct pointers?

Reply:
Because we can't unpack them in the union the same way because they'll mess up the size/representation

Reply:
Oh, yes you are right.

Reply:
Changed the implementation at the safeposix side.